azure-sink-connector/src/integration-test/java/io/aiven/kafka/connect/azure/ConnectRunner.java

```diff
@@ -0,0 +1,118 @@
+import org.apache.kafka.connect.runtime.ConnectorConfig;
+import org.apache.kafka.connect.runtime.Herder;
+import org.apache.kafka.connect.runtime.Worker;
+import org.apache.kafka.connect.runtime.isolation.Plugins;
+import org.apache.kafka.connect.runtime.rest.RestServer;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorInfo;
+import org.apache.kafka.connect.runtime.standalone.StandaloneConfig;
+import org.apache.kafka.connect.runtime.standalone.StandaloneHerder;
+import org.apache.kafka.connect.storage.MemoryOffsetBackingStore;
+import org.apache.kafka.connect.util.Callback;
+import org.apache.kafka.connect.util.FutureCallback;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class ConnectRunner {
+    private static final Logger LOG = LoggerFactory.getLogger(ConnectRunner.class);
+
+    private final File pluginDir;
+    private final String bootstrapServers;
+    private final int offsetFlushInterval;
+
+    private Herder herder;
+    private Connect connect;
+
+    public ConnectRunner(final File pluginDir, final String bootstrapServers, final int offsetFlushIntervalMs) {
+        this.pluginDir = pluginDir;
+        this.bootstrapServers = bootstrapServers;
+        this.offsetFlushInterval = offsetFlushIntervalMs;
+    }
+
+    void start() {
+        final Map<String, String> workerProps = new HashMap<>();
+        workerProps.put("bootstrap.servers", bootstrapServers);
+
+        workerProps.put("offset.flush.interval.ms", Integer.toString(offsetFlushInterval));
+
+        // These don't matter much (each connector sets its own converters), but need to be filled with valid classes.
+        workerProps.put("key.converter", "org.apache.kafka.connect.converters.ByteArrayConverter");
+        workerProps.put("value.converter", "org.apache.kafka.connect.converters.ByteArrayConverter");
+        workerProps.put("internal.key.converter", "org.apache.kafka.connect.json.JsonConverter");
+        workerProps.put("internal.key.converter.schemas.enable", "false");
+        workerProps.put("internal.value.converter", "org.apache.kafka.connect.json.JsonConverter");
+        workerProps.put("internal.value.converter.schemas.enable", "false");
+
+        // Don't need it since we'll memory MemoryOffsetBackingStore.
+        workerProps.put("offset.storage.file.filename", "");
+
+        workerProps.put("plugin.path", pluginDir.getPath());
+
+        final Time time = Time.SYSTEM;
+        final String workerId = "test-worker";
+        final String kafkaClusterId = "test-cluster";
+
+        final Plugins plugins = new Plugins(workerProps);
+        final StandaloneConfig config = new StandaloneConfig(workerProps);
+        final Worker worker = new Worker(workerId, time, plugins, config, new MemoryOffsetBackingStore());
+        herder = new StandaloneHerder(worker, kafkaClusterId);
+
+        final RestServer rest = new RestServer(config);
+
+        connect = new Connect(herder, rest);
+        connect.start();
```

Comment:
Oh goodness this is depending on so many internal APIs. I would much rather us just use the EmbeddedConnect class from upstream. We'll have to fix this at some point...